### PR TITLE
Make release checklist more precise and add workflow for unit testing

### PR DESCRIPTION
**Pull request by insipx**

### CODE REVIEW

File: RELEASE-CHECKLIST.md
Comment (dvdplm) [inline RELEASE-CHECKLIST.md:14]: 👍

**Review by dvdplm — APPROVED** [review]

File: substrate-archive/src/lib.rs
Comment (TarikGul) [inline substrate-archive/src/lib.rs:70]: Nice1

**Review by TarikGul — APPROVED** [review]
LGTM

**Comment by insipx** [comment]
Don't merge yet, might add more changes to improve the release pipeline

**Comment by insipx** [comment]
Thoughts about experience releasing archive today and possible improvements and issues I've found with it:

Mostly i'm trying to think of a way to make it easier to review/automate/make releases. I think we should try to keep the diff to merge back into master as small as possible, it would be very easy to unintentionally commit something that would lead to more PR's fixing things on release days. IE this diff: https://github.com/paritytech/substrate-archive/pull/289 seems not ideal

I think I should change the release checklist to be more like:

- Make a PR to update CHANGELOG and bump versions of projects to master/
- pull in `master` from `release/`, update versions to latest polkadot with `diener`
- make a PR for the release against the `release` branch

I spent today researching different release methods ([this](https://martinfowler.com/articles/branching-patterns.html#release-branch) article was particularly informative) however I don't see an easy way we can both: 

1. maintain releases to be frozen with the latest substrate/polkadot version 
2. also have master follow `substrate/` master

That isn't the `release branch` method. However we're going to have to be extra-careful that release and master don't diverge too much, and so we should modify the `release`branch as little as possible.

Not sure if anyone else has thoughts on this. I've edited the release-checklist to hopefully solve this issue for the next release

**Comment by insipx** [comment]
I'd appreciate another skim of the new `RELEASE-CHECKLIST`before merging, but other than that it's ready. modified the workflows to work with the new branching strategy. I think it'll make things easier to manage going forward.

Workflows are tested in `v0.0.0-testpleaseignore` which will be here once the CI finishes: https://github.com/paritytech/substrate-archive/releases/tag/v0.0.0-testpleaseignore (took a few tries to make sure everything was working as expected)

for some reason the unit tests fail where previously they would pass -- despite all that is changing is markdown :shrug:

**Comment by dvdplm** [comment]
> another skim of the new `RELEASE-CHECKLIST`before merging

lgtm

**Pull request merged**
